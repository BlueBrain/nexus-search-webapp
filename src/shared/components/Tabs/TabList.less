.ellipsis {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
.tab-element-container {
  display: flex;
  flex-direction: row;
  .tab-item {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    text-align: left;
    max-width: 250px;
    > .title {
      font-size: 1.2em;
    }
    > .description {
      white-space: normal;
      line-height: 1.2em;
      color: rgba(0, 0, 0, 0.5);
    }
  }
  .edit-button-container {
    display: flex;
    width: 40px;
  }
}
