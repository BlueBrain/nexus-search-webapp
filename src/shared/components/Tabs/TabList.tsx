import * as React from 'react';
import './TabList.less';
import { Tabs } from 'antd';
import TabElement from './TabElement';

const { TabPane } = Tabs;

type TabItem = {
  id: string;
  label: string;
  description: string;
};

type TabListProps = {
  items: TabItem[];
  onSelected: (activeKey: string) => void;
  defaultActiveId?: string;
  activeKey?: string;
  position?: 'left' | 'right' | 'top' | 'bottom' | undefined;
  tabAction?: React.ReactElement;
  editButton?: (id: string) => React.ReactNode;
};

const TabList: React.FunctionComponent<TabListProps> = ({
  items,
  onSelected,
  defaultActiveId,
  activeKey,
  position = 'left',
  children,
  tabAction,
  editButton,
}) => {
  return (
    <div className="tab-list">
      <Tabs
        tabPosition={position}
        onChange={onSelected}
        defaultActiveKey={defaultActiveId}
        activeKey={activeKey}
        tabBarExtraContent={tabAction}
      >
        {items.map(({ label, description, id }) => (
          <TabPane
            tab={
              <TabElement
                label={label}
                description={description}
                id={id}
                editButton={editButton}
              />
            }
            key={id}
          >
            {children}
          </TabPane>
        ))}
      </Tabs>
    </div>
  );
};

export default TabList;
