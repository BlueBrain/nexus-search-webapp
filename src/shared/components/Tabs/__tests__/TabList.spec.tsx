import * as React from 'react';
import { shallow } from 'enzyme';

import TabList from '../TabList';

const tabListComponent = <TabList items={[]} onSelected={jest.fn} />;
const wrapper = shallow(tabListComponent);

describe('TabList component', () => {
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

jest.restoreAllMocks();
