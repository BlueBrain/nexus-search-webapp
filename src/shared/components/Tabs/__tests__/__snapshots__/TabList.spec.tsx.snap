// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabList component should render correctly 1`] = `
<div
  className="tab-list"
>
  <Tabs
    hideAdd={false}
    onChange={[Function]}
    tabPosition="left"
  />
</div>
`;
